src/lib/recurringPurposes.js

```diff
@@ -367,7 +368,37 @@ const queryRecurringTimeseriesWithCloseStartOrEnd = async (
   return results || []
 }
 
-// Similar trips = close start/end point
+export const filterTripsBasedOnDistance = (timeseries, baseDistance) => {
+  const maxDistance =
+    baseDistance + baseDistance * TRIPS_DISTANCE_SIMILARITY_RATIO
+  const minDistance =
+    baseDistance - baseDistance * TRIPS_DISTANCE_SIMILARITY_RATIO
+  return timeseries.filter(ts => {
+    return (
+      ts.aggregation?.totalDistance >= minDistance &&
+      ts.aggregation?.totalDistance <= maxDistance
+    )
+  })
+}
+
+const postFilterResults = (results, timeserie, { oldPurpose }) => {
```

Comment:
let's add a JSDoc.

Reply:
Benefits: you've a typed variables everywhere. Here in your postFilterResults, you don't know what you are handling. How I know that results is an array of timeseries? And that a timeserie as an `_id` attribute? It's way easier to read / edit the code within our IDE if it knows what type are used. You have autocomplete and so on. No need to open cozy-doctypes to know the attributes list for instance ;)

JSDoc is here for typing, not only for documentation ;)

Reply:
So yes, there is a huge benefit to have jsdoc almost everywhere. Do you prefer JSDoc or TS? ;)

Reply:
This is why, when you're not in a TS file, you can/should type it anyway 😘 

> Caveat: it's heavy and slow things down to do it every time ;)

You'll find a way to reduce this time ;). 

For instance you can do something like: 


```
 * @typedef {import('./types').TimeseriesGeoJSON} TimeSeries 
```

at the top of your file, and then only use `TimeSeries` instead of having to import the type everytime. 

> much more than with type annotation in python, for instance.

can you give me an exemple?